board/designa/snappermx6/ksz9896.c

```diff
@@ -0,0 +1,215 @@
+ #define KSZ9896_MMDAADR                                        0x0E
+
+  //PHY MMD Setup register
+ #define KSZ9896_MMDACR_FUNC                                    0xC000
+ #define KSZ9896_MMDACR_FUNC_ADDR                               0x0000
+ #define KSZ9896_MMDACR_FUNC_DATA_NO_POST_INC                   0x4000
+ #define KSZ9896_MMDACR_FUNC_DATA_POST_INC_RW                   0x8000
+ #define KSZ9896_MMDACR_FUNC_DATA_POST_INC_W                    0xC000
+ #define KSZ9896_MMDACR_DEVAD                                   0x001F
+
+// PHY status register
+#define KSZ9896_PHY_BASIC_STATUS								0x0102
+#define KSZ9896_PHY_LINK_PARTNER_STATUS							0x010A
+#define KSZ9896_PHY_1000BT_STATUS								0x0114
+
+ //MMD LED Mode register
+ #define KSZ9896_MMD_LED_MODE_LED_MODE                          0x0010
+ #define KSZ9896_MMD_LED_MODE_LED_MODE_TRI_COLOR_DUAL           0x0000
+ #define KSZ9896_MMD_LED_MODE_LED_MODE_SINGLE                   0x0010
+ #define KSZ9896_MMD_LED_MODE_RESERVED                          0x000F
+ #define KSZ9896_MMD_LED_MODE_RESERVED_DEFAULT                  0x0001
+  
+ //MMD EEE Advertisement register
+ #define KSZ9896_MMD_EEE_ADV_1000BT_EEE_EN                      0x0004
+ #define KSZ9896_MMD_EEE_ADV_100BT_EEE_EN                       0x0002
+
+
+ //KSZ9896 MMD registers
+ #define KSZ9896_MMD_LED_MODE                                   0x02, 0x00
+ #define KSZ9896_MMD_EEE_ADV                                    0x07, 0x3C
+
+static int ksz9896_write16(struct udevice *dev, uint8_t port, uint16_t reg, uint16_t val)
+{
+	uint8_t data[2] = {val >> 8, val};
+	if (dm_i2c_write(dev, port << 12 | reg, data, 2)) {
+		printf("cannot write 0x%x=0x%04x\n", reg, val);
+		return -1;
+	}
+	return 0;
+}
+
+static int ksz9896_read16(struct udevice *dev, uint8_t port, uint16_t reg)
+{
+	uint16_t data;
+	if (dm_i2c_read(dev, port << 12 | reg, (uint8_t *)&data, 2)) {
+		return -1;
+	}
+	return __be16_to_cpu(data);
+}
+
+static uint32_t ksz9896_read32(struct udevice *dev, uint8_t port, uint16_t reg)
+{
+	uint32_t val;
+	if (dm_i2c_read(dev, port << 12 | reg, (uint8_t *)&val, sizeof(val))) {
+		return -1;
+	}
+    return __be32_to_cpu(val);
+}
+
+static int ksz9896_write_mmd(struct udevice *dev, uint8_t port, uint8_t dev_addr, uint16_t reg_addr, uint16_t data)
+{
+    //Select register operation
+    ksz9896_write16(dev, port, KSZ9896_MMDACR,
+       KSZ9896_MMDACR_FUNC_ADDR | (dev_addr & KSZ9896_MMDACR_DEVAD));
+  
+    //Write MMD register address
+    ksz9896_write16(dev, port, KSZ9896_MMDAADR, reg_addr);
+  
+    //Select data operation
+    ksz9896_write16(dev, port, KSZ9896_MMDACR,
+       KSZ9896_MMDACR_FUNC_DATA_NO_POST_INC | (dev_addr & KSZ9896_MMDACR_DEVAD));
+  
+    //Write the content of the MMD register
+    ksz9896_write16(dev, port, KSZ9896_MMDAADR, data);
+
+	return 0;
+}
+
+int ksz9896_init(void)
+{
+	struct udevice *dev;
+	uint32_t id = 0;
+    uint8_t xmii_ctrl_1 = 0x18;
+
+	if (i2c_get_chip_for_busnum(0, 0x5f, 2, &dev) < 0)
```

Comment:
You should be able to use the phy subsystem

Compatible string is marvell,88E1510

U_BOOT_PHY_DRIVER() to declare it

I'm not sure that the phy subsystem does driver model very well though...